src/freertos_drivers/esp_idf/EspIdfWiFi.hxx

```diff
@@ -767,28 +603,13 @@ private:
     unsigned scanResultsEntryIndex_ {INT_MAX};
     std::string staConnectPass_; ///< last station connect attempt password
     const std::string hostname_; ///< published hostname
-    int mdnsStaLockCount_; ///< counter for recursive mDNS STA lock
     uint8_t apClientCount_; ///< number of connected wifi clients
```

Comment:
During my testing, I've seen `WIFI_EVENT_AP_STADISCONNECTED` raised when the AP was not running as we shift over from APSTA (to support network scanning, no AP running or started) to STA only mode. We likely should explore and augment the logic around this variable when we have not seen AP running prior to receiving the event.

Reply:
@atanisoft Please have another look at this logic in `wifi_event_handler()`. I've made some updates. look for `ipAcquiredAp_`, `apClientCount_`, and `apLeased_`.

Reply:
I don't find any references to `apLeased_` with a search, but I do find the other two so I'm guessing it may be a misreference? The event will still be raised by ESP-IDF but your code will ignore it. I likely missed it but I also don't see these vars declared